### PR TITLE
Fix compare zips empty uploaded packs

### PR DESCRIPTION
**Pull request by ilaner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/utils/compare_staging_graph.py [hunk 1]
```diff
@@ -140,6 +140,37 @@ def file_diff(output_path: Path, zip1_files: str, zip2_files: str, file: str, di
...
+    index_graph_path: Path,
+    collected_packs_id_set: Path,
+    collected_packs_graph: Path,
+    message: list[str],
+):
+    output_path.mkdir(exist_ok=True, parents=True)
+    # compare directories
+    dir_cmp = filecmp.dircmp(zip_id_set, zip_graph)
+    dir_cmp.report_full_closure()
+    message.append(f"Zip difference for {output_path}")
```
Comment (tkatzir) [inline Tests/scripts/utils/compare_staging_graph.py:158]: What if there are no differences?

File: Tests/scripts/utils/compare_staging_graph.py [hunk 2]
```diff
@@ -140,6 +140,37 @@ def file_diff(output_path: Path, zip1_files: str, zip2_files: str, file: str, di
...
+    for file in dir_cmp.common_files:
+        pack = file.removesuffix(".zip")
+        if diff_files := compare_zips(zip_id_set / file, zip_graph / file, output_path / pack):
+            message.append(f'Detected differences in the following files for pack {pack}: {", ".join(diff_files)}')
+    if compare_indexes(index_id_set_path, index_graph_path, output_path):
```
Comment (tkatzir) [inline Tests/scripts/utils/compare_staging_graph.py:163]: In the signature of this method (`compare`), the `output_path` param is first.
Here (`compare_indexes`) it's the last.
Can the params order be standardized?

File: Tests/scripts/utils/compare_staging_graph.py [hunk 3]
```diff
@@ -140,6 +140,37 @@ def file_diff(output_path: Path, zip1_files: str, zip2_files: str, file: str, di
...
+        message.append("Detected differences between index.json files")
```
Comment (tkatzir) [inline Tests/scripts/utils/compare_staging_graph.py:164]: Can the differences be listed in the message?

File: Tests/scripts/utils/compare_staging_graph.py [hunk 4]
```diff
@@ -140,6 +140,37 @@ def file_diff(output_path: Path, zip1_files: str, zip2_files: str, file: str, di
...
+    if file_diff_text(output_path / "collect_tests_diff.json", collected_packs_id_set, collected_packs_graph):
+        message.append("Detected differences between collect tests results")
```
Comment (tkatzir) [inline Tests/scripts/utils/compare_staging_graph.py:166]:
```suggestion
        message.append("Detected differences in collect tests results")
```

Comment (tkatzir) [inline Tests/scripts/utils/compare_staging_graph.py:166]: Can the differences be listed in the message?

File: Tests/scripts/utils/compare_staging_graph.py (hunk 2 above)
Comment (ilaner) [reply Tests/scripts/utils/compare_staging_graph.py:163]: now output_path is last in all methods

File: Tests/scripts/utils/compare_staging_graph.py (hunk 3 above)
Comment (ilaner) [reply Tests/scripts/utils/compare_staging_graph.py:166]: The difference could be too big for a slack message

File: Tests/scripts/utils/compare_staging_graph.py (hunk 1 above)
Comment (ilaner) [reply Tests/scripts/utils/compare_staging_graph.py]: fixed

File: Tests/scripts/utils/compare_staging_graph.py (hunk 3 above)
Comment (tkatzir) [reply Tests/scripts/utils/compare_staging_graph.py:166]: Can it be in an attachment?
Maybe in the log, with a link to it?

Comment (tkatzir) [reply Tests/scripts/utils/compare_staging_graph.py:166]: Good enough. A (human) debugger could do a diff themselves...

**Review by tkatzir — APPROVED** [review]

**Pull request merged**